src/qibolab/serialize.py

```diff
@@ -123,6 +167,45 @@ def load_instrument_settings(
     return instruments
 
 
+def _dump_pulse(pulse: Pulse):
+    data = asdict(pulse)
+    if pulse.type in (PulseType.FLUX, PulseType.COUPLERFLUX):
+        del data["frequency"]
+    if "shape" in data:
+        data["shape"] = str(pulse.shape)
+    data["type"] = data["type"].value
+    if "channel" in data:
+        del data["channel"]
+    if "relative_phase" in data:
+        del data["relative_phase"]
+    return data
+
+
+def _dump_single_qubit_natives(natives: SingleQubitNatives):
+    data = {}
+    for fld in fields(natives):
+        pulse = getattr(natives, fld.name)
+        if pulse is not None:
+            data[fld.name] = _dump_pulse(pulse)
+            del data[fld.name]["qubit"]
+    return data
```

Comment:
Here it would be really nice to do just `SingleQubitNatives(**natives)`.

This is currently not possibly for two reasons:
1. the true one is that you have to `del` `"qubit"`, not sure why
2. the contingent one is that dataclasses do not support proper nested deserialization - but Pydantic does!

Reply:
> 1. the true one is that you have to `del` `"qubit"`, not sure why

I think this is only for cosmetic reasons, because in the runcard we do something like
```yaml
single_qubit:
    0:
        RX:
              # pulse details here
```
(now in json) so it would be redundant to have the `qubit` in the pulse details since these are already under the qubit. We could put it back by updating all the runcards, but even better in #792 we will hopefully drop it completely from the `Pulse` so this issue will disappear.